src/platforms/generic_unix/lib/sys.c

```diff
@@ -550,8 +553,67 @@ Context *sys_create_port(GlobalContext *glb, const char *driver_name, term opts)
 
 term sys_get_info(Context *ctx, term key)
 {
-    UNUSED(ctx);
-    UNUSED(key);
+    if (globalcontext_is_term_equal_to_atom_string(ctx->global, key, ATOM_STR("\x7", "os_type"))) {
+        struct utsname buf;
+        errno = 0;
+        if (UNLIKELY(uname(&buf) != 0)) {
+            return ERROR_ATOM;
+        }
+
+        const char *sys_name;
+
+        if (strcmp(buf.sysname, "Linux") == 0) {
```

Comment:
readability pro-tip: let's factor out this block to a static `sysname_to_atom` helper function, no changes are needed than taking this out.